samples/net/sockets/echo_async/README.rst

```diff
@@ -59,7 +59,8 @@ Running application on POSIX Host
 
 The same application source code can be built for a POSIX system, e.g.
 Linux. (Note: if you look at the source, you will see that the code is
-the same except the header files are different for Zephyr vs POSIX.)
+the same except the header files are different for Zephyr vs POSIX, and
+there're additional option to set for Linux to make a socket IPv6-only).
```

Comment:
Fixed.

That option is actually Linux-specific, not part of POSIX. At the beginning of this paragraph it says "a POSIX system, e.g. Linux", so hopefully even a reader not closely familiar with "family tree" of the OSes won't be too confused how it all relates together. (And for someone familiar, the above gives "absolutely true" picture without simplifications.)